src/Sylius/Bundle/AdminBundle/EventListener/AdminFilterSubscriber.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\EventListener;
+
+use Sylius\Bundle\AdminBundle\Storage\FilterStorageInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+final class AdminFilterSubscriber implements EventSubscriberInterface
+{
+    public function __construct(
+        private FilterStorageInterface $filterStorage,
+    ) {
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            KernelEvents::REQUEST => 'onKernelRequest',
+        ];
+    }
+
+    public function onKernelRequest(RequestEvent $event): void
+    {
+        if (!$this->isMainRequest($event)) {
+            return;
+        }
+
+        if ('html' !== $event->getRequest()->getRequestFormat()) {
+            return;
+        }
+
+        $eventRequest = $event->getRequest();
```

Comment:
minor thing but it can be before line 41 and you will be able to use this variable in the second `if`